### PR TITLE
AzureRT: Fix Get-AzureVM

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

File: src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs [hunk 1]
```diff
@@ -101,14 +101,18 @@ protected override void ExecuteCommand()
         private List<T> GetVMContextList<T>(string serviceName, NSM.DeploymentGetResponse deployment)
             where T : PVM.PersistentVMRoleContext, new()
         {
-            var vmRoles = new List<NSM.Role>(deployment.Roles.Where(
-                r => string.IsNullOrEmpty(Name)
-                  || r.RoleName.Equals(Name, StringComparison.InvariantCultureIgnoreCase)));
+            Func<NSM.Role, bool> typeMatched =
+                r => string.Equals(r.RoleType, PersistentVMRoleStr, StringComparison.OrdinalIgnoreCase);
```
Comment (AzureRT) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs:105]: Why is it StringComparison.OrdinalIgnoreCase instead of StringComparison.InvariantCultureIgnoreCase ?

File: src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs [hunk 2]
```diff
@@ -101,14 +101,18 @@ protected override void ExecuteCommand()
...
+            Func<NSM.Role, bool> nameMatched =
+                r => string.IsNullOrEmpty(this.Name) || r.RoleName.Equals(this.Name, StringComparison.InvariantCultureIgnoreCase);
+
+            var vmRoles = new List<NSM.Role>(deployment.Roles.Where(r => typeMatched(r) && nameMatched(r)));
+
+            return CreateVMContextList<T>(serviceName, deployment, vmRoles);
         }
 
-        private List<T> GetVMContextList<T>(string serviceName, NSM.DeploymentGetResponse deployment, List<NSM.Role> vmRoles)
+        private List<T> CreateVMContextList<T>(string serviceName, NSM.DeploymentGetResponse deployment, List<NSM.Role> vmRoles)
```
Comment (AzureRT) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs:115]: Why did you change this method name?

File: src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs (hunk 1 above)
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs:105]: I think we are only handling ASCII characters, so I think the `StringComparison.OrdinalIgnoreCase` one is enough.

File: src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs (hunk 2 above)
Comment (huangpf) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/PersistentVMs/GetAzureVM.cs:115]: I would like to fix the function name so that it matches what it's doing. `CreateVMContextList` calls `CreateVMContext` and it's creating a new list of `T` objects, based on the input of `NSM.Role` (which also indicates maybe the verb shall be `Convert` or `Map`), where as the previous one `GetVMContextList` does mean "getting a list of VM context objects from the deployment.

**Comment by huangpf** [comment]
retest this please
